### DISCUSSION TITLE
Setup of a Motomini (or any other yaskawa robot arm) with Moveit 2

### DISCUSSION THREAD

**Discussion by ivanrulik** [Yaskawa-Global/motoros2#269] [Q&A]
In this [repo](https://github.com/ivanrulik/motomini_moveit_ws) I have placed a small project to both learn and make a clean step by step of how to set up any supported yaskawa robot arm with moveit 2.

While I’m trying to reach this level of [application](https://github.com/YaskawaEurope/ros2-starter-for-yaskawa-robots) I’m taking the advice from @gavanderhoorn of doing the steps one by one to familiarize with every tool.

Currently I’m having issues with the launch files from the MoveIt setup assistant. As I’m trying to launch the demo.launch.py generated by the moveit tool.

```bash
ros2 launch motomini_moveit_config demo.launch.py 
[INFO] [launch]: All log files can be found below /home/ivan/.ros/log/2024-06-24-14-26-22-441461-ivan-Zephyrus-G16-86447
[INFO] [launch]: Default logging verbosity is set to INFO
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead.
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Expected 'value' to be one of [<class 'float'>, <class 'int'>, <class 'str'>, <class 'bool'>, <class 'bytes'>], but got '()' of type '<class 'tuple'>'
```

Any advice or help on this is greatly appreciated

**Comment by gavanderhoorn** [comment]
Unfortunately this looks like an issue on the MoveIt side. I'm actually not sure what causes the exception you included.

One thing I would recommend though: unless you have a very good reason to, I'd not use groups, but chains in your SRDF. Chain would go from `base_link` to `tool0`. Chains have IK solvers which will let you use the interactive marker in RViz, and allow MoveIt to accept Cartesian goals (ie: `Pose`s).

**Comment by gavanderhoorn** [comment]
Have you been able to get things to work @ivanrulik?

**Reply by ivanrulik** [reply]
Hi @gavanderhoorn. Not yet, I've been busy using YMConnect for an internal project but I do want to re-visit this project and the usage of MotoROS2.
